### ISSUE TITLE
Does it guarantee to converge?

### ISSUE THREAD

**Issue by w3ntao** [chncyhn/flappybird-qlearning-bot#6]
I deploy your strategy to my program:

```
//update when death:

Q[lastS][lastA] = (1 - lrAlpha) * Q[lastS][lastA] + lrAlpha * deadReward;

for all (S, A) in reversed(expHistory):
    Q[S][A] = (1 - lrAlpha) * Q[S][A] + lrAlpha * (aliveReward + gamma * max(Q[S_]));
    S_ = S;
```

My bird hit highest score at about 500 round, then scores keep going down. At about 1000 round, it never make through more than 1 pipe and keep hitting ceiling or upper pipe.

Have you encounter such problem?

![jkqlt](https://user-images.githubusercontent.com/6308795/42929916-c23224ea-8b3b-11e8-80b2-6bafe5245537.png)

**Comment by chncyhn** [comment]
Hello @WentaoZero, I don't remember having this problem.
  
Does this happen consistently for all runs? Also have you tried using different hyperparameters (learning rate etc.) and discretization levels?

**Comment by w3ntao** [comment]
Hi @chncyhn , good to have you replied.
That was a week ago, now I already have it fixed.
This is my repo: [https://github.com/WentaoZero/q-bird](https://github.com/WentaoZero/q-bird)
And your work is cited 😄

**Comment by chncyhn** [comment]
@WentaoZero That's great to hear! Very nice project. 👍
